### PR TITLE
[ fix #718 ] Fix unsolved metas in propositional uniqueness

### LINKED ISSUE #718: Unsolved metas when using Data.List.Relation.Unary.Unique.Propositional
I suspect it's the same problem as #683

Test case:

```agda
open import Data.Bool
open import Data.List.Base
open import Data.List.Relation.Unary.Unique.Propositional

record AtMostTwo : Set where
  constructor _,_
  field list   : List Bool
        unique : Unique list

_ : AtMostTwo
_ = [] , []
```

Edit:

The fact that this is a valid `[] {a} {A} {a} {a} {A}` instantiation of the
constructors' implicits tells you exactly what the problem is: the constructors
are re-exported without being applied.

### PR DESCRIPTION
**Pull request by gallais**
This is a possible fix for #718 adopting the same strategy we used in the Sublist case.

### CODE REVIEW

File: src/Data/List/Relation/Unary/AllPairs.agda [hunk 1]
```diff
@@ -23,65 +26,54 @@ import Relation.Nullary.Decidable as Dec
...
 
--- AllPairs R xs means that every pair of elements (x , y) in xs is a
--- member of relation R (as long as x comes before y in the list).
-
-infixr 5 _∷_
-
-data AllPairs {a ℓ} {A : Set a} (R : Rel A ℓ) : List A → Set (a ⊔ ℓ) where
-  []  : AllPairs R []
-  _∷_ : ∀ {x xs} → All (R x) xs → AllPairs R xs → AllPairs R (x ∷ xs)
+open import Data.List.Relation.Unary.AllPairs.Core public
```
Comment (gallais) [inline src/Data/List/Relation/Unary/AllPairs.agda]: This unqualified re-export in a `{a ℓ} {A : Set a} {R : Rel A ℓ}` context looks a bit dodgy though.

**Comment by MatthewDaggitt** [comment]
The question I want to get to the bottom of, is why is this not necessary for propositional membership? Why do `here` and `there` work just fine while `[]` and `_::_` don't?

**Comment by gallais** [comment]
I still thing the unqualified open public is dodgy.

**Comment by MatthewDaggitt** [comment]
> I still think the unqualified open public is dodgy.

Thanks, I forgot about that. Looked at it again, and short of agda/agda#3210 being fixed I'm not sure there's a whole lot we can do? I guess we could re-parameterise the `AllPairs` type, but we can't re-parameterise the constructors without losing the ability to pattern match on them. (I've tried using pattern synonyms for this, but doesn't seem to work).

It's definitely better than it was before, so unless you have any cases where you know that it breaks(?), I propose merging it in.

**Pull request merged**